odbc/src/api/data.rs

```diff
@@ -186,6 +193,8 @@ impl IntoCData for Bson {
                 if out > i64::MAX as f64 || out < i64::MIN as f64 {
                     Err(ODBCError::IntegralTruncation(out.to_string()))
                 } else {
+                    // This is safe because we've already checked that the value is within the range of i64
+                    #[allow(clippy::cast_possible_truncation)]
```

Comment:
Well, using TryFrom in data will result in me changing most of the conversion functions in their entirety. I looked into doing that initially, but stopped because it adds a lot of diff. Happy to do so though.